File: dawii_20222_miercoles_backend_grupo_04-master/src/main/java/com/cibertec/service/SalaService.java
package com.cibertec.service;

import java.util.List;
import java.util.Optional;

import com.cibertec.entity.Sala;

public interface SalaService {

	
	public Sala InsertaSala(Sala obj);
	public Sala InsertaActualizaSala(Sala obj);
	public abstract void EliminaSala(int idSala);
	public List<Sala> ListaSala();
	public List<Sala> listaSalaporNumero(String Numero);
	Optional<Sala> findByIdSala(int idSala);
	
	public List<Sala> listaSalapornumeroAlumnosSede(String numero, int numAlumnos, int idSede, int estado);

}

File: dawii_20222_miercoles_frontend_grupo_04-master/src/app/services/sala.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { AppSettings } from '../app.settings';
import { Sala } from '../models/sala.model';

const baseUrlSala = AppSettings.API_ENDPOINT+ '/sala';
const baseUrlSalaCrud = 'http://localhost:8090/rest/crudSala';

@Injectable({
  providedIn: 'root'
})
export class SalaService {

  constructor(private http: HttpClient) { }

    registraSala(obj:Sala): Observable<any> {
      return this.http.post(baseUrlSala, obj);
    
    }
    listaSala(numero:string, numAlumnos:number, idSede: number, estado : number): Observable<any>{
        const params = new HttpParams().set("numero", numero).set("numAlumnos", numAlumnos).set("idSede", idSede).set("estado", estado);  
        return this.http.get(baseUrlSala + "/listaSalaConParametros", {params});
      }

      consultaPorNumeroCrud(filtro:string):Observable<Sala[]>{
        return  this.http.get<Sala[]>(baseUrlSalaCrud +"/listaSalaporNumero/"+filtro); 
    }  
  
    insertaCrud(obj:Sala):Observable<any>{
        return this.http.post(baseUrlSalaCrud + "/registraSala", obj);
    }
  
    actualizaCrud(obj:Sala):Observable<any>{
        return this.http.put(baseUrlSalaCrud + "/actualizaSala", obj);
    }
  
    eliminaCrud(idSala:number):Observable<any>{
        return this.http.delete(baseUrlSalaCrud + "/eliminaSala/"+ idSala);
    }
      
  }

